### DISCUSSION TITLE
Synchronizing selected Explorer and TreeView items

### DISCUSSION THREAD

**Discussion by DmitryOffsec** [microsoft/vscode-discussions#631] [Extension Development QnA] [answered]
I'm using a treeView that allows me to get a simplified view of the file structure that can be seen in [Explorer](https://github.com/Security-Experts-Community/vscode-xp/blob/b809889ebea618ccfac20e4290b1a48fcfdd7e81/client/src/views/contentTree/contentTreeProvider.ts#L264). I would like to get the functionality that allows you to synchronize the selected item from the Explorer and the treeView. I tried to use the reveal method:
```
vscode.workspace.onDidOpenTextDocument(
	async (td: vscode.TextDocument) => {
		const fileName = td.fileName;
		if(!fileName.endsWith(".co")) {
			return;
		}

		const correlationDirPath = path.dirname(fileName);
		const explorerCorrelation = await Correlation.parseFromDirectory(correlationDirPath);
		await kbTree.reveal(explorerCorrelation,
			{
				focus: true,
				expand: true,
				select: true
			});
	}
);
```
But that doesn't work. In this case, it should be taken into account that the selected treeItem can be located on any nesting of the treeView. How can I make it so that if a file is opened in Explorer, the corresponding treeItem will be selected in the treeView?

**Comment by r3m0t** [comment]
Documents can be opened without editors so I would suggest listening to `onDidChangeTabs`.

Did you implement `getParent `? https://code.visualstudio.com/api/references/vscode-api#TreeDataProvider&lt;T&gt;

**Reply by DmitryOffsec** [reply]
[Yes](https://github.com/Security-Experts-Community/vscode-xp/blob/b809889ebea618ccfac20e4290b1a48fcfdd7e81/client/src/views/contentTree/contentTreeProvider.ts#L407)

**Reply by r3m0t** [reply]
Log when it's called and what it returns?

**Reply by DmitryOffsec** [reply, marked as answer]
I was debugging the `getParent `method and realized that it was not returning the same elements as the `getChildren `method. After fixing this functionality has earned.
There is only one question left, how can I select the desired element in the treeView, but avoid showing this treeView? I want the Explorer window to remain open in the Activity Bar, but when switching to the TreeView, the desired element is selected there. I experimented with various options for the reveal method, but the tree still shows up.
